### DISCUSSION TITLE
"You don't have write permissions" error when attempting to install RuboCop

### DISCUSSION THREAD

**Discussion by xobs** [ruby/rubygems#6893] [Q&A]
I was told by RuboCop (https://github.com/rubocop/rubocop/issues/12130) that this is a bug in Rubygems.

I tried to follow the documentation on how to install RuboCop, but the instructions failed. The instructions on https://rubocop.org/ said to run:

```
$ gem install rubocop
```

However, this did not work on my setup:

```
$ gem install rubocop
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.6.0 directory.
$
```

I thought maybe there were additional setup instructions, such as maybe creating an environment, installing a different interpreter, or specifying an installation path, but they said that there weren't any additional instructions. They said that it was an issue with rubygems.

This is a default install on Mac using the system interpreter.

**Comment by deivid-rodriguez** [comment]
This is because of using Mac default Ruby installation. It installs gems to a location that's not writable. The error message is accurate. Maybe some day they will remove it or make it more friendly. For now, I suggest to use `asdf` or similar version manager to install Ruby. Cheers!
